### PR TITLE
fix(query): check valid warn message (#9919)

### LINKED ISSUE #9919: _warningRow is not iterable
## What you are doing?

when i config sequelize log with winston logger with config below:

``` javascript
// 标准日志服务组件...
let wsLogs = require(global.appRoot + '/libs/WsLogs');
let sqlLogger = wsLogs.logger('sql');

let sequelizeOptions = {
    operatorsAliases: false,
    //operatorsAliases: Op,
    // 显示 SQL 警告信息...
    showWarnings: true,
    //logging: false,
    //logging: sqlLogger.verbose,
    logging: (msg) => sqlLogger.verbose(msg),
    // disable inserting undefined values as NULL
    // - default: false
    omitNull: false,
    ...db_config[name]
};
```
I had set options with `showWarnings: true` to show sql errors,

## What do you expect to happen?
when i exe a warning sql,it crashed with error: `_warningRow is not iterable`,and can't show any warning msg from mysql,how to work around this?
i hava exe this sql in terminal manual:

``` sql
mysql> INSERT INTO `tours` (`id`,`name`,`city`,`thumb`,`phone1`,`phone2`,`body`,`detail_address`,`traffic_blue`,`traffic_self`,`privilege`,`is_recommend`,`coord_x`,`coord_y`,`tips`,`open_time`,`close_time`,`url`,`level`,`toursn`,`price`,`tey_limits`,`created_at`,`updated_at`) VALUES (10029,'测试景区[10029][F:0][R:0][YL:100][D:0]','xh','http://ekt-userface.oss-cn-shenzhen.aliyuncs.com/20180502/ea4ae9c477c9aa2e415fda6745a7d3b5.jpg','','','','','','','',0,0,0,'','13:30','22:00','http://www.mltc.cn/',0,10029,188,0,'2018-05-01 12:30:00','2018-09-13 16:01:19');
Query OK, 1 row affected, 1 warning (0.01 sec)

mysql> show warnings;
+---------+------+-------------------------------------------+
| Level   | Code | Message                                   |
+---------+------+-------------------------------------------+
| Warning | 1265 | Data truncated for column 'name' at row 1 |
+---------+------+-------------------------------------------+
1 row in set (0.01 sec)
```
this is because column name is too long.

## What is actually happening?
my environment: 
**sequelize version** `4.38.0`;
**mysql version** `5.7.12`;
**winston version** `3.1.0`;

### PR DESCRIPTION
**Pull request by tsasaki609**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change
fix #9919 
validate that warning row is iterable

### CODE REVIEW

**Comment by tsasaki609** [comment]
Oops I forgot. I made a simple unit test.
Thanks for the review.

File: lib/dialects/mysql/query.js [hunk 1]
```diff
@@ -178,6 +178,7 @@ class Query extends AbstractQuery {
       const warningMessage = 'MySQL Warnings (' + (this.connection.uuid||'default') + '): ';
       const messages = [];
       for (const _warningRow of warningResults) {
+        if (typeof _warningRow[Symbol.iterator] !== 'function') break;
```
Comment (sushantdhiman) [inline lib/dialects/mysql/query.js]: `continue` instead of `break`

File: test/unit/dialects/mysql/query.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Comment (sushantdhiman) [inline test/unit/dialects/mysql/query.test.js]: I would recommend integration test for this, you can use this https://github.com/sequelize/sequelize/issues/9919#issuecomment-420938123 or create your own. (Model.create part is important)

Create a new file called `warning.test.js` [here](https://github.com/sequelize/sequelize/tree/master/test/integration/dialects/mysql)

1. Move this test https://github.com/sequelize/sequelize/blob/master/test/integration/sequelize.test.js#L305
2. Add a new one for `Model.create` case

**Comment by tsasaki609** [comment]
I implemented an integration test.

**Comment by sushantdhiman** [comment]
Thanks @tsasaki609

**Pull request merged**